contracts/oracles/common/CappedOracle.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.25;
+
+import { OracleInterface } from "../../interfaces/OracleInterface.sol";
+import { ensureNonzeroAddress, ensureNonzeroValue } from "@venusprotocol/solidity-utilities/contracts/validators.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { CappedOracle } from "./CappedOracle.sol";
+import { Transient } from "../../lib/Transient.sol";
+
+/**
+ * @title CappedOracle
+ * @notice This oracle fetches the price of a correlated token and caps the growth rate
+ */
+abstract contract CappedOracle is OracleInterface {
+    using Transient for bytes32;
+
+    /// Slot to cache the asset's price, used for transient storage
+    bytes32 public constant CACHE_SLOT = keccak256(abi.encode("venus-protocol/oracle/common/CappedOracle/cache"));
+
+    //// @notice Growth rate percentage in seconds. Ex: 1e18 is 100%
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable GROWTH_RATE_PER_SECOND;
+
+    /// @notice Snapshot update interval
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SNAPSHOT_INTERVAL;
+
+    /// @notice Last stored snapshot price
+    uint256 public snapshotPrice;
+
+    /// @notice Last stored snapshot timestamp
+    uint256 public snapshotTimestamp;
+
+    /// @notice Emitted when the snapshot is updated
+    event SnapshotUpdated(uint256 price, uint256 timestamp);
+
+    /// @notice Constructor for the implementation contract.
```

Comment:
I would add natspec for each argument, mentioning that `snapshotInterval` set to zero means that caps are disabled